### PR TITLE
DevEx: Add applicationContext to DocketRecord constructor

### PR DESCRIPTION
**Pull request by ericsorenson**

### CODE REVIEW

File: shared/src/business/entities/DocketRecord.js [hunk 1]
```diff
@@ -10,7 +10,14 @@ const { getAllEventCodes } = require('../../utilities/getAllEventCodes');
...
  * @constructor
  */
-function DocketRecord(rawDocketRecord) {
+function DocketRecord(rawDocketRecord, { applicationContext }) {
+  if (!applicationContext) {
+    throw new TypeError('applicationContext must be defined');
+  }
+
+  this.filedBy = rawDocketRecord.filedBy;
+  this.editState = rawDocketRecord.editState;
```
Comment (rachaelparris) [inline shared/src/business/entities/DocketRecord.js]: Are these lines duplicated below?

File: shared/src/business/entities/DocketRecord.test.js [hunk 2]
```diff
@@ -1,75 +1,109 @@
...
+            {
+              eventCode: 'O',
+              filingDate: new Date('2000-01-01').toISOString(),
+              index: 0,
+            },
+            { applicationContext },
+          ).isValid(),
+        ).toBeFalsy();
+      },
+      { applicationContext },
```
Comment (rachaelparris) [inline shared/src/business/entities/DocketRecord.test.js]: This line is unnecessary I believe?

File: shared/src/business/entities/cases/Case.js [hunk 3]
```diff
@@ -226,16 +226,18 @@ function Case(rawCase, { applicationContext }) {
   if (!applicationContext) {
     throw new TypeError('applicationContext must be defined');
   }
+
   this.associatedJudge = rawCase.associatedJudge || Case.CHIEF_JUDGE;
-  this.automaticBlocked = rawCase.automaticBlocked;
-  this.automaticBlockedDate = rawCase.automaticBlockedDate;
-  this.automaticBlockedReason = rawCase.automaticBlockedReason;
+  this.caseNote = rawCase.caseNote;
```
Comment (kkoskelin) [inline shared/src/business/entities/cases/Case.js]: as long as you've got other changes to do, perhaps these can be re-sorted again? :)

**Review by rachaelparris — APPROVED** [review]
A couple minor things but otherwise 👍

**Review by kkoskelin — APPROVED** [review]
Once comments are addressed 👍

**Pull request merged**